dynamic_instrumentation/src/instrument.rs

```diff
@@ -286,7 +513,10 @@ impl<'tcx> Visitor<'tcx> for CollectInstrumentationPoints<'_, 'tcx> {
                     .dest(&dest)
                     .add_to(self);
             }
-            Rvalue::Ref(_, bkind, p) if has_outer_deref(p) => {
+            Rvalue::Ref(_, bkind, p)
+                if has_outer_deref(p)
+                    && is_region_or_unsafe_ptr(place_ty(&remove_outer_deref(*p, self.tcx()))) =>
+            {
```

Comment:
I can't seem to find which commit added this change. `git log -p dynamic_instrumentation/src/instrument.rs` makes it look like it comes from `7675d10cdf0d3885ab5656d3882b111548ac0407`, but that commit isn't listed as part of this PR... I don't understand what's happening here--maybe merge weirdness?

The history here is really convoluted and I would love to see a simple sequence of commits without merges.

Reply:
Anyhow, the reason I wanted to know where this line came from is because I'd like to understand better exactly which bug is being fixed by this change, which should ideally be in the commit message of a small commit that does only this.

Reply:
> The history here is really convoluted and I would love to see a simple sequence of commits without merges.

Yeah, that gets hard on these very large PRs that are developed for months.  I think any opportunity we see to split something out into a separate, more targeted PR, we should do.

Reply:
I think rebasing is the only thing that makes sense for long-living PRs. Addressed comments going stale isn't a huge loss, and the ability for reviewers to figure out what comes from where when reviewing code is critical.